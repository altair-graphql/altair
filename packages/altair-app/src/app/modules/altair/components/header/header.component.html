<div
  class="header__logo-wrapper"
  [ngClass]="{ 'header__logo-wrapper--experimental': experimentalEnabled }"
>
  <img
    src="assets/img/altair_logo.svg"
    alt="logo"
    class="header__logo"
  >
</div>

<app-window-switcher
  [windows]="windows"
  [windowIds] = "windowIds"
  [activeWindowId]="activeWindowId"
  [closedWindows]="closedWindows"
  [isElectron]="isElectron"
  (newWindowChange)="newWindowChange.emit($event)"
  (activeWindowChange)="activeWindowChange.emit($event)"
  (removeWindowChange)="removeWindowChange.emit($event)"
  (duplicateWindowChange)="duplicateWindowChange.emit($event)"
  (windowNameChange)="windowNameChange.emit($event)"
  (repositionWindowChange)="repositionWindowChange.emit($event)"
  (reopenClosedWindowChange)="reopenClosedWindowChange.emit($event)"
></app-window-switcher>

<div class="header__actions">
  <ul class="header__menu">
    <li
      *ngFor="let panel of headerPanels trackBy:trackById"
      (click)="togglePanelActiveChange.emit(panel)"
      [popper]="panel.title"
      [popperPlacement]="'bottom'"
      class="header__menu-item"
      [ngClass]="{ 'header__menu-item--active': panel.isActive }"
    >
      <app-icon name="cpu"></app-icon>
      <nz-modal
        [nzVisible]="panel.isActive"
        (nzVisibleChange)="togglePanelActiveChange.emit(panel)"
        [nzTitle]="modalTitle"
        [nzContent]="modalContent"
        [nzFooter]="null"
        (nzOnCancel)="togglePanelActiveChange.emit(panel)"
      >
        <ng-template #modalTitle>
          <div class="app-dialog-header">
            <div class="app-dialog-title">{{ panel.title }}</div>
          </div>
        </ng-template>

        <ng-template #modalContent>
          <app-element-wrapper
            [element]="panel.element"
          ></app-element-wrapper>
        </ng-template>
      </nz-modal>
    </li>

    <li
      nz-dropdown
      nzTrigger="click"
      [nzDropdownMenu]="environmentsMenu"
      class="header__menu-item"
    >
      <span *ngIf="activeEnvironment">
        <app-icon name="eye"></app-icon>
        <span>{{ activeEnvironment?.title }}</span>
      </span>
      <span *ngIf="activeEnvironment === undefined">
        <app-icon name="eye-off"></app-icon>
        <span>{{ 'NO_ENVIRONMENT_TEXT' | translate }}</span>
      </span>
    </li>
    <nz-dropdown-menu #environmentsMenu="nzDropdownMenu">
      <ul nz-menu class="environments-dropdown__list">
        <ng-container *ngIf="environments?.subEnvironments.length">
          <li
            nz-menu-item
            (click)="selectActiveEnvironmentChange.emit(environment.id)"
            *ngFor="let environment of environments.subEnvironments; trackBy:trackById;"
            [ngClass]="{ 'active': environment.id === activeEnvironment }"
          >
            {{ environment.title }}
          </li>
          <li
            nz-menu-item
            (click)="selectActiveEnvironmentChange.emit(undefined)"
          >
            {{ 'NO_ENVIRONMENT_TEXT' | translate }}
          </li>
          <li nz-menu-divider></li>
        </ng-container>
        <li
          nz-menu-item
          (click)="toggleEnvironmentManagerChange.emit(true)"
        >
          {{ 'ENVIRONMENTS_BUTTON' | translate }}...
        </li>
      </ul>
    </nz-dropdown-menu>
    <li
      nz-dropdown
      nzTrigger="click"
      [nzDropdownMenu]="settingsMenu"
      class="header__menu-item"
    >
      <app-icon name="settings"></app-icon>
    </li>
    <nz-dropdown-menu #settingsMenu="nzDropdownMenu">
      <ul nz-menu>
        <li nz-menu-item (click)="importWindowChange.emit()">{{ 'IMPORT_WINDOW_TEXT' | translate }}</li>
        <li nz-menu-item (click)="showImportCurlDialogChange.emit(true)">{{ 'IMPORT_WINDOW_FROM_CURL_TEXT' | translate }}</li>
        <li nz-menu-item (click)="showSettingsDialogChange.emit()">{{ 'SETTINGS_TEXT' | translate }}</li>
        <li nz-menu-item (click)="openLink($event, 'https://github.com/altair-graphql/altair')">{{ 'STAR_ON_GITHUB_TEXT' | translate }}</li>
        <li nz-menu-item (click)="openLink($event, 'https://github.com/altair-graphql/altair/issues/new?template=Bug_report.md')">{{ 'REPORT_BUG_TEXT' | translate }}</li>
        <li nz-menu-item (click)="exportBackupDataChange.emit()">{{ 'EXPORT_BACKUP_DATA_TEXT' | translate }}</li>
        <li nz-menu-item (click)="importBackupDataChange.emit()">{{ 'IMPORT_BACKUP_DATA_TEXT' | translate }}</li>
      </ul>
    </nz-dropdown-menu>
  </ul>
</div>
