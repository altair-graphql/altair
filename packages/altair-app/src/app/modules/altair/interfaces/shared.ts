export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
