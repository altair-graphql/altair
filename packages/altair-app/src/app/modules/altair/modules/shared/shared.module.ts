import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
