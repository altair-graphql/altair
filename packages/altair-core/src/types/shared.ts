export type IDictionary<V = any> = Record<string, V>;

export type TODO = any;

// export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export interface BaseDocument {
  id: string;
  created_at?: number;
  updated_at?: number;
}

export interface BaseOwnableDocument extends BaseDocument {
  ownerUid: string;
}

type TypedOmit<T, K extends keyof T> = Omit<T, K>;
export type CreateDTO<
  T extends BaseDocument | BaseOwnableDocument
> = T extends BaseOwnableDocument
  ? TypedOmit<T, keyof BaseOwnableDocument>
  : TypedOmit<T, keyof BaseDocument>;
export type UpdateDTO<T extends BaseDocument | BaseOwnableDocument> = CreateDTO<
  T
> &
  Pick<T, 'id'>;
