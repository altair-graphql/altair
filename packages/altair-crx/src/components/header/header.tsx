export const Header = () => {
  return <div className="devtools-header">Header content</div>;
};
