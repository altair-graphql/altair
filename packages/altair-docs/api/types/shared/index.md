# types/shared

## Index

| Member | Description |
| :------ | :------ |
| [BaseDocument](interfaces/BaseDocument.md) | - |
| [BaseOwnableDocument](interfaces/BaseOwnableDocument.md) | - |
| [CreateDTO](type-aliases/CreateDTO.md) | - |
| [IDictionary](type-aliases/IDictionary.md) | - |
| [NoExtraProperties](type-aliases/NoExtraProperties.md) | - |
| [TODO](type-aliases/TODO.md) | - |
| [UpdateDTO](type-aliases/UpdateDTO.md) | - |
