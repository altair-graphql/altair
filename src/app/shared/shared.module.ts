import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { TranslateModule } from '@ngx-translate/core';
import { NgxElectronModule } from 'ngx-electron';
import { SortablejsModule } from 'angular-sortablejs';
import { NgPipesModule } from 'ngx-pipes';
import { ContextMenuModule } from 'ngx-contextmenu';
import { DndModule } from '@beyerleinf/ngx-dnd';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';


@NgModule({
  imports: [
    CommonModule,
    TranslateModule.forRoot(),
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    ContextMenuModule.forRoot(),
    DndModule.forRoot(),
  ],
  exports: [
    TranslateModule,
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    ContextMenuModule,
    DndModule,
    NguiAutoCompleteModule,
  ]
})
export class SharedModule { }
