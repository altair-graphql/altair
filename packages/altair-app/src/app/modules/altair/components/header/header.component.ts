import { Component, EventEmitter, Input, Output } from '@angular/core';
import { AltairPanel } from 'altair-graphql-core/build/plugin/panel';
import {
  EnvironmentsState,
  EnvironmentState,
} from 'altair-graphql-core/build/types/state/environments.interfaces';
import { PerWindowState } from 'altair-graphql-core/build/types/state/per-window.interfaces';
import { SettingsState } from 'altair-graphql-core/build/types/state/settings.interfaces';
import { WindowState } from 'altair-graphql-core/build/types/state/window.interfaces';
import { externalLink } from '../../utils';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styles: [],
})
export class HeaderComponent {
  @Input() windows: WindowState = {};
  @Input() windowIds: string[] = [];
  @Input() closedWindows: PerWindowState[] = [];
  @Input() activeWindowId = '';
  @Input() isElectron = false;
  @Input() headerPanels: AltairPanel[] = [];
  @Input() activeEnvironment?: EnvironmentState;
  @Input() environments?: EnvironmentsState;
  @Input() settings?: SettingsState;
  @Output() activeWindowChange = new EventEmitter<string>();
  @Output() newWindowChange = new EventEmitter();
  @Output() removeWindowChange = new EventEmitter<string>();
  @Output() duplicateWindowChange = new EventEmitter<string>();
  @Output() windowNameChange = new EventEmitter<{
    windowId: string;
    windowName: string;
  }>();
  @Output() repositionWindowChange = new EventEmitter<{
    currentPosition: number;
    newPosition: number;
  }>();
  @Output() reopenClosedWindowChange = new EventEmitter();
  @Output() togglePanelActiveChange = new EventEmitter<AltairPanel>();
  @Output() selectActiveEnvironmentChange = new EventEmitter<
    string | undefined
  >();
  @Output() toggleEnvironmentManagerChange = new EventEmitter<boolean>();
  @Output() showSettingsDialogChange = new EventEmitter();
  @Output() importWindowChange = new EventEmitter();
  @Output() showImportCurlDialogChange = new EventEmitter<boolean>();
  @Output() exportBackupDataChange = new EventEmitter();
  @Output() importBackupDataChange = new EventEmitter();

  constructor() {}

  openLink(e: Event, url: string) {
    externalLink(e, url);
  }

  trackById<T extends { id?: string }>(index: number, item: T) {
    return item.id;
  }
}
