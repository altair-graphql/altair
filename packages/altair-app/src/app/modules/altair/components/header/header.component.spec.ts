import { NO_ERRORS_SCHEMA } from '@angular/core';
import { expect, describe, it } from '@jest/globals';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MockModule } from 'ng-mocks';
import { mount, NgxTestWrapper } from '../../../../../testing';
import { SharedModule } from '../../modules/shared/shared.module';

import { HeaderComponent } from './header.component';

describe('HeaderComponent', () => {
  let wrapper: NgxTestWrapper<HeaderComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: HeaderComponent,
      imports: [MockModule(SharedModule)],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should create', () => {
    expect(wrapper.componentInstance).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
