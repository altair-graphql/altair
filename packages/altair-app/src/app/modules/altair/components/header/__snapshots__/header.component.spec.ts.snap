// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render correctly 1`] = `
<app-header>
  <div
    class="header__logo-wrapper"
  >
    <img
      alt="logo"
      class="header__logo"
      src="assets/img/altair_logo.svg"
    />
  </div>
  <app-window-switcher />
  <div
    class="header__actions"
  >
    <ul
      class="header__menu"
    >
      
      <li
        class="header__menu-item"
        nz-dropdown=""
        nztrigger="click"
      >
        
        <span>
          <app-icon
            name="eye-off"
          />
          <span>
            
          </span>
        </span>
        
      </li>
      
      <nz-dropdown-menu>
        <ul
          class="environments-dropdown__list"
          nz-menu=""
        >
          
          <li
            nz-menu-item=""
          >
             ... 
          </li>
          
        </ul>
        
      </nz-dropdown-menu>
      <li
        class="header__menu-item"
        nz-dropdown=""
        nztrigger="click"
      >
        <app-icon
          name="settings"
        />
      </li>
      
      <nz-dropdown-menu>
        <ul
          nz-menu=""
        >
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
          <li
            nz-menu-item=""
          >
              
          </li>
          
        </ul>
        
      </nz-dropdown-menu>
    </ul>
  </div>
</app-header>
`;
